app/Actions/User/Notify.php

```diff
@@ -19,7 +19,7 @@ public function do(Photo $photo): void
 		}
 
 		// The admin is always informed
-		$users = new Collection([User::query()->find(0)]);
+		$users = new Collection(User::query()->find(0));
```

Comment:
Why that change?

I am not even sure that this change is correct. The constructor checks if the passed value is an instance of `Arrayable` and the calls `toArray()` on the passed value. `User` is a model and a such an instance of `Arrayable`. However `User::toArray()` returns the array representation of a user not an array with a single user object.

This looks really wrong.

Reply:
```
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   app/Actions/User/Notify.php                                                                                                                                                                                           
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  25     Parameter #1 ...$values of method Illuminate\Support\Collection<int,Illuminate\Database\Eloquent\Model|null>::push() expects Illuminate\Database\Eloquent\Model|null, Illuminate\Database\Eloquent\Collection given.  
 ------ ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
 ```

Reply:
But have you tried, if it actually works? The error by Larastan/PHPStan seems to be a false positive.

The return type of `Eloquent::find` is a union type, see [Laravel API 9.x: `Illuminate/Database/Eloquent/Builder::find()`](https://laravel.com/api/9.x/Illuminate/Database/Eloquent/Builder.html#method_find):

 - `Builder::find()` may return a `Collection` if multiple results are found
 - `Builder::find()` returns a single instance of `Model` if queried for an ID as we do here

However, it seems that the Larastan stubs (wrongly) assume that `find` always returns a `Collection` and hence PHPStan is complaining about the additional brackets, because `[]` around a `Collection` would indeed make no sense as it would create a collection of a collection.

I haven't tried to run the code yet, but this is what I conjecture from code inspection.

Reply:
Add a deeper look at it and agree with your point. Revered.

Reply:
Fixed in [25db8c3](https://github.com/LycheeOrg/Lychee/pull/1469/commits/25db8c3a0f74087e089b3d677ffea1c66e742b60)